vpr/src/draw/draw.cpp

```diff
@@ -1043,2427 +672,176 @@ void init_draw_coords(float width_val) {
-            }
-        }
-    }
-}
+    rr_type = rr_graph.node_type(rr_node);
+    i = rr_graph.node_xlow(rr_node); /* NB: Global rr graphs must have only unit */
+    j = rr_graph.node_ylow(rr_node); /* length channel segments.                 */
 
-static void drawnets(ezgl::renderer* g) {
-    t_draw_state* draw_state = get_draw_state_vars();
-    t_draw_coords* draw_coords = get_draw_coords_vars();
-    /* This routine draws the nets on the placement.  The nets have not *
-     * yet been routed, so we just draw a chain showing a possible path *
-     * for each net.  This gives some idea of future congestion.        */
+    switch (rr_type) {
+        case CHANX:
+            return (chanx_track[i][j]);
 
-    ClusterBlockId b1, b2;
-    auto& cluster_ctx = g_vpr_ctx.clustering();
+        case CHANY:
+            return (chany_track[i][j]);
 
-    float NET_ALPHA = draw_state->net_alpha;
-
-    g->set_line_dash(ezgl::line_dash::none);
-    g->set_line_width(0);
-
-    /* Draw the net as a star from the source to each sink. Draw from centers of *
-     * blocks (or sub blocks in the case of IOs).                                */
-
-    for (auto net_id : cluster_ctx.clb_nlist.nets()) {
-        if (cluster_ctx.clb_nlist.net_is_ignored(net_id))
-            continue; /* Don't draw */
-
-        g->set_color(draw_state->net_color[net_id],
-                     draw_state->net_color[net_id].alpha * NET_ALPHA);
-        b1 = cluster_ctx.clb_nlist.net_driver_block(net_id);
-        ezgl::point2d driver_center = draw_coords->get_absolute_clb_bbox(b1,
-                                                                         cluster_ctx.clb_nlist.block_type(b1))
-                                          .center();
-        for (auto pin_id : cluster_ctx.clb_nlist.net_sinks(net_id)) {
-            b2 = cluster_ctx.clb_nlist.pin_block(pin_id);
-            ezgl::point2d sink_center = draw_coords->get_absolute_clb_bbox(b2,
-                                                                           cluster_ctx.clb_nlist.block_type(b2))
-                                            .center();
-            g->draw_line(driver_center, sink_center);
-            /* Uncomment to draw a chain instead of a star. */
-            /* driver_center = sink_center;  */
-        }
+        default:
+            vpr_throw(VPR_ERROR_OTHER, __FILE__, __LINE__,
+                      "in get_track_num: Unexpected node type %d for node %d.\n",
+                      rr_type, inode);
+            return OPEN;
     }
 }
 
-static void draw_congestion(ezgl::renderer* g) {
-    /* Draws all the overused routing resources (i.e. congestion) in various contrasting colors showing congestion ratio.   */
+/* This helper function determines whether a net has been highlighted. The highlighting
+ * could be caused by the user clicking on a routing resource, toggled, or
+ * fan-in/fan-out of a highlighted node.
+ */
+bool draw_if_net_highlighted(ClusterNetId inet) {
```

Comment:
If there is no corresponding comment for this function in the .h file, duplicate or summarize this comment there (we're moving toward commenting what a function does in the .h file, and how it works in the .cpp file if necessary).